File: 135022193-main/2024/Done/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    int check;
    int digit;
    int split;
    int sum;
    int sum1 = 0;
    int sum2 = 0;
    int i = 0;
    long amex;
    long master;

    long card = get_long("Number: ");
    long visa = card;
    long copyCard = card;

    while (card != 0)
    {
        digit = card % 10;
        if (i % 2 == 0)
        {
            sum1 += digit;
        }
        else
        {
            digit = digit * 2;
            if (digit >= 10)
            {
                while (digit != 0)
                {
                    split = digit % 10;
                    sum2 += split;
                    digit /= 10;
                }
            }
            sum2 += digit;
        }
        card /= 10;
        i++;
    }
    sum = sum1 + sum2;
    check = sum % 10;
    if (check == 0)
    {
        while (visa >= 10)
            {
                visa /= 10;
            }
            if ((visa == 4 && i == 13) || (visa == 4 && i == 16))
            {
                printf("VISA\n");
            }
        else if (i == 15)
        {
            amex = copyCard;
            while (amex >= 100)
            {
                amex /= 10;
            }
            if (amex == 34 || amex == 37)
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (i == 16)
        {
            master = copyCard;
            while (master >= 100)
            {
                master /= 10;
            }
            if (master > 50 && master < 56)
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: 135022193-main/2024/Done/sentimental-credit/credit.py
from cs50 import get_string, get_int


def main():

    i = 0
    sum1 = 0
    sum2 = 0
    card = get_int("number: ")
    cardCopy = card

# create the sums for checking cards validity
    while card != 0:
        digit = card % 10
        if i % 2 == 0:
            sum2 += digit
        elif i % 2 != 0:
            digit = digit * 2
            if digit >= 10:
                tmp = list(str(digit))
                for j in range(len(tmp)):
                    digit = int(tmp[j])
                    sum1 += digit
            else:
                sum1 += digit

        card //= 10
        i += 1

    sum = sum1 + sum2
    tmp = list(str(sum))

    # check if sums last digit is 0
    if tmp[len(tmp) - 1] != '0':
        print("INVALID")
        return 0

# declare variables for checking validity
    VISA = cardCopy
    while (VISA >= 10):
        VISA //= 10

    AMEX = cardCopy
    while (AMEX >= 100):
        AMEX //= 10

    MASTERCARD = cardCopy
    while (MASTERCARD >= 100):
        MASTERCARD //= 10

    # check validity of VISA
    if i == 16 and VISA == 4 or i == 13 and VISA == 4:
        print("VISA")
        return 0

    # check validity of AMEX
    elif i == 15 and AMEX == 34 or AMEX == 37:
        print("AMEX")
        return 0

    # check validity of MASTERCARD
    elif i == 16 and MASTERCARD >= 51 and MASTERCARD <= 55:
        print("MASTERCARD")
        return 0

    else:
        print("INVALID")


main()
